src/data/mv_polynomial/basic.lean

```diff
@@ -297,10 +298,18 @@ lemma is_id (f : mv_polynomial σ R →+* mv_polynomial σ R)
   f p = p :=
 hom_eq_hom f (ring_hom.id _) hC hX p
 
+@[ext] lemma alg_hom_ext' {A B : Type*} [comm_semiring A] [comm_semiring B]
+  [algebra R A] [algebra R B] {f g : mv_polynomial σ A →ₐ[R] B}
+  (h₁ : f.comp (is_scalar_tower.to_alg_hom R A (mv_polynomial σ A)) =
+        g.comp (is_scalar_tower.to_alg_hom R A (mv_polynomial σ A)))
+  (h₂ : ∀ i, f (X i) = g (X i)) : f = g :=
+alg_hom.coe_ring_hom_injective (mv_polynomial.ring_hom_ext'
+  (congr_arg alg_hom.to_ring_hom h₁) h₂)
```

Comment:
Does it make sense to prove this for `add_monoid_algebra` too?

Reply:
My view on this is that it would be better if it was defined for `add_monoid_algebra`, but that we also shouldn't enforce maximum generality on everything, because `add_monoid_algebra` is rarely used and it's a lot of effort to duplicate everything on `polynomial` and `mv_polynomial`.